urllib3/response.py

```diff
@@ -327,9 +360,15 @@ def read(self, amt=None, decode_content=None, cache_content=False):
                     # no harm in redundantly calling close.
                     self._fp.close()
                     flush_decoder = True
+                    if (self.strict_content_length and self.length not in (0, None)):
+                        # httplib requires an iterator/string instead of count of bytes
+                        data = [0 for x in range(self._fp_bytes_read)]
```

Comment:
Added custom exception in 2015f77, it has slightly different semantics (taking int instead of iterator as first arg). This seems like it might cause confusion but avoids creating a possibly huge list on large streams just to reuse the httplib exception. I'm +.5 on this and open to alternatives I'm missing.